src/ucp/core/ucp_ep.c

```diff
@@ -357,11 +362,18 @@ ucs_status_t ucp_worker_create_mem_type_endpoints(ucp_worker_h worker)
 void ucp_worker_destroy_mem_type_endpoints(ucp_worker_h worker)
 {
     ucs_memory_type_t mem_type;
+    ucp_ep_h ep;
 
     ucs_memory_type_for_each(mem_type) {
-        if (worker->mem_type_ep[mem_type] != NULL) {
-           ucp_ep_destroy_internal(worker->mem_type_ep[mem_type]);
-           worker->mem_type_ep[mem_type] = NULL;
+        ep = worker->mem_type_ep[mem_type];
+        if (ep != NULL) {
+            ucs_debug("memtype ep %p: destroy", ep);
+            ucs_assert(ep->flags & UCP_EP_FLAG_INTERNAL);
+            ucs_assert(ucp_ep_ext_control(ep)->local_ep_id == (ucs_ptr_map_key_t)ep);
+
+            ucp_ep_cleanup_lanes(ep);
+            ucp_ep_delete(ep);
+            worker->mem_type_ep[mem_type] = NULL;
```

Comment:
are we not calling ucp_ep_destroy_internal() because of local_ep_id?
can we set local_ep_id==UCP_EP_ID_INVALID for memtype endpoint?

Reply:
```ucp_ep_destroy_internal``` is just makes decision about should it be destroyed gracefully or just free memory. for mem_type EPs we know the right way

```set local_ep_id==UCP_EP_ID_INVALID``` will cause assert in ```ucp_ep_delete->ucp_ep_local_id(ep)```

Reply:
maybe can remove some code duplication if reusing ucp_ep_destroy_internal?

Reply:
there is only duplication is call ucp_ep_cleanup_lanes, all other asserts/calls are actual
I tried to add ```is_internal``` parameter to ```ucp_ep_destroy_internal``` call, but it require much more code update & debug for possible stacks for EP destroy (or always check for ```ep->flags & UCP_EP_FLAG_INTERNAL``` for every call)

Reply:
ok